dix/dixfonts.c

```diff
@@ -745,7 +746,12 @@ doListFontsAndAliases(ClientPtr client, LFclosurePtr c)
         .sequenceNumber = client->sequence
     };
 
-    char *bufferStart = calloc(1, rep.length << 2);
+    size_t bufsize = (size_t)rep.length << 2;
```

Comment:
The warning was because `rep.length` is unsigned vs `size_t` being a signed int?
We will hardly hit that limit here, not sure if it's worth adding the check.
We can just do `size_t bufsize = (size_t)(stringLens + nnames);` and use it below I guess.

Also if @metux is going to rewrite this using his newly introduced buffer structure maybe we should just wait for it.

Reply:
If we consider this a bug, then it should be done separately from the rewrite - but better move to a separate length variable and write the final value into the struct later.